core/src/main/java/org/kiwix/kiwixmobile/core/utils/AlertDialogShower.kt

```diff
@@ -19,16 +19,28 @@
 package org.kiwix.kiwixmobile.core.utils
 
 import android.app.Activity
+import android.app.Dialog
 import androidx.appcompat.app.AlertDialog
-import org.kiwix.kiwixmobile.core.utils.KiwixDialog.StartHotspotManually
 import javax.inject.Inject
 
 class AlertDialogShower @Inject constructor(private val activity: Activity) : DialogShower {
   override fun show(dialog: KiwixDialog, vararg clickListeners: () -> Unit) {
-    AlertDialog.Builder(activity)
+    create(dialog, *clickListeners)
+  }
+
+  override fun create(dialog: KiwixDialog, vararg clickListeners: () -> Unit): Dialog {
+    return AlertDialog.Builder(activity)
       .apply {
         dialog.title?.let(this::setTitle)
-        setMessage(activity.getString(dialog.message, *bodyArguments(dialog)))
+        dialog.icon?.let(this::setIcon)
+        dialog.message?.let {
+          setMessage(
+            activity.getString(
```

Comment:
sometimes the formatter does this where it agressively splits lines, I think `setMessage(...)` will fit within our line limitations